code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,305 @@
+/obj/structure/closet/crate/mail/proc/populate(amount)
+	var/mail_count = min(amount, storage_capacity)
+	// Fills the
+	var/list/mail_recipients = list()
+
+	for(var/mob/living/carbon/human in GLOB.player_list)
+		if(human.stat == DEAD || !human.mind)
+			continue
+		// Skip wizards, nuke ops, cyborgs; Centcom does not send them mail
+		if(human.faction.Find("Station", 1, 0) != 0)
+			continue
+
+		mail_recipients += human.mind
+
+	for(var/i in 1 to mail_count)
+		var/obj/item/mail/new_mail
+		if(prob(FULL_CRATE_LETTER_ODDS))
+			new_mail = new /obj/item/mail(src)
+		else
+			new_mail = new /obj/item/mail/envelope(src)
+
+		var/datum/mind/recipient = pick_n_take(mail_recipients)
+		if(recipient)
+			new_mail.initialize_for_recipient(recipient)
+		else
+			new_mail.junk_mail()
+
+	update_icon()
+
+/// Crate for mail that automatically depletes the economy subsystem's pending mail counter.
+/obj/structure/closet/crate/mail/economy/Initialize(mapload)
+	. = ..()
+	populate(SSeconomy.mail_waiting)
+	SSeconomy.mail_waiting = 0
+
+/// Crate for mail that automatically generates a lot of mail. Usually only normal mail, but on lowpop it may end up just being junk.
+/obj/structure/closet/crate/mail/full
+	name = "brimming mail crate"
+	desc = "A certified post crate from CentCom. Looks stuffed to the gills."
+
+/obj/structure/closet/crate/mail/full/Initialize(mapload)
+	. = ..()
+	populate(INFINITY)
+
+
+/// Mailbag.
+/obj/item/storage/bag/mail
+	name = "mail bag"
+	desc = "A bag for letters, envelopes, and other postage."
+	icon = 'icons/obj/library.dmi'
+	icon_state = "bookbag"
+	//worn_icon_state = "bookbag"
+	resistance_flags = FLAMMABLE
+
+/obj/item/storage/bag/mail/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/storage = GetComponent(/datum/component/storage)
+	storage.max_w_class = WEIGHT_CLASS_NORMAL
+	storage.max_combined_w_class = 42
+	storage.max_items = 21
+	storage.display_numerical_stacking = FALSE
+	storage.set_holdable(list(
+		/obj/item/mail,
+		/obj/item/smallDelivery,
+		/obj/item/paper
+	))
+
+/obj/item/paper/fluff/junkmail_redpill
+	name = "smudged paper"
+	icon_state = "scrap"
+	var/nuclear_option_odds = 0.1
+
+/obj/item/paper/fluff/junkmail_redpill/Initialize(mapload)
+	. = ..()
+	if(!prob(nuclear_option_odds)) // 1 in 1000 chance of getting 2 random nuke code characters.
+		info = "<i>You need to escape the simulation. Don't forget the numbers, they help you remember:</i> '[rand(0,9)][rand(0,9)][rand(0,9)]...'"
+		return
+	var/code = random_nukecode()
+	for(var/obj/machinery/nuclearbomb/selfdestruct/self_destruct in GLOB.nuke_list)
+		self_destruct.r_code = code
+	message_admins("Through junkmail, the self-destruct code was set to \"[code]\".")
+	info = "<i>You need to escape the simulation. Don't forget the numbers, they help you remember:</i> '[code[rand(1,5)]][code[rand(1,5)]]...'"
```

Comment:
This gives 2 digits if the numbers are legit, and 3 if not, therefore allowing you to meta if your numbers are real. Is this intentional?